src/middlewared/middlewared/plugins/consulkv.py

```diff
@@ -112,6 +138,31 @@ class ConsulService(Service):
 
         return await self.reload()
 
+    def _aws_config_file(self, region=None, key_id=None, access_key=None):
+        config_path = '/root/.aws/config'
+        credentials_path = '/root/.aws/credentials'
+        aws_vault = '/root/.aws/'
```

Comment:
Any chance we can override /root/.aws path? I am skeptical about writing to boot media. Maybe it could be written to /etc/ and use etc.py plugin for this?